detectable/src/main/java/com/synopsys/integration/detectable/detectables/swift/lock/parse/PackageResolvedDataChecker.java

```diff
@@ -0,0 +1,40 @@
+package com.synopsys.integration.detectable.detectables.swift.lock.parse;
+
+import java.util.Arrays;
+
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detectable.detectables.swift.lock.data.v2.PackageResolvedV2;
+import com.synopsys.integration.log.IntLogger;
+import com.synopsys.integration.log.Slf4jIntLogger;
+
+// The concern is Path packages and anything new Apple introduces: https://developer.apple.com/documentation/swift_packages/package/dependency/3197884-package
+public class PackageResolvedDataChecker {
+    private final IntLogger logger;
+
+    protected static final String[] KNOWN_KINDS = { "remoteSourceControl" };
+
+    public PackageResolvedDataChecker() {
+        this(new Slf4jIntLogger(LoggerFactory.getLogger(PackageResolvedDataChecker.class)));
+    }
+
+    public PackageResolvedDataChecker(IntLogger logger) {
+        this.logger = logger;
+    }
+
+    public void logUnknownPackageTypes(PackageResolvedV2 packageResolved) {
+        packageResolved.getPackages().stream()
+            .filter(packageItem -> packageItem.getKind()
```

Comment:
If KNOWN_KINDS was a list, would .filter(packageItem -> !KNOWN_KINDS.contains(packageItem.getKind())) be equivalent?  This filtering seems more complicated than it needs to be.

Reply:
The kind is Optional which is where most of the complexity comes from.
The check `!KNOWN_KINDS.contains(packageItem.getKind())) ` would always return true.
Going from an Array to a List just changes line 28 from
`.map(foundKind -> Arrays.stream(KNOWN_KINDS)`
to `.map(foundKind -> KNOWN_KINDS.stream()`

Reply:
This whole class is a totally optional step just to provide some better logging in the event something goes wrong.
If it's too complex, it's probably better to bin the whole thing.

Reply:
Ah okay.

Reply:
I don't think it's egregiously complex, I just thought it was more complex than it had to be but sounds like that's not the case